cmd/config.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/Azure/kubectl-az/cmd/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var configCmd = &cobra.Command{
+	Use:   "config",
+	Short: "Configure kubectl-az",
+}
+
+func init() {
+	rootCmd.AddCommand(configCmd)
+	configCmd.AddCommand(unsetCmd)
+	configCmd.AddCommand(showCmd)
+
+	configCmd.AddCommand(nodeCmd)
+	nodeCmd.Flags().String(utils.NodeKey, "", "Kubernetes node name.")
+
+	configCmd.AddCommand(resourceIDCmd)
+	resourceIDCmd.Flags().String(utils.ResourceIDKey, "",
+		`Resource ID containing all information of the VMSS instance using format:
+		e.g. /subscriptions/mySubID/resourceGroups/myRG/providers/myProvider/virtualMachineScaleSets/myVMSS/virtualMachines/myInsID.
+		Notice it is not case sensitive.`,
+	)
+
+	configCmd.AddCommand(vmmsCmd)
+	vmmsCmd.Flags().String(utils.SubscriptionIDKey, "", "Subscription ID.")
+	vmmsCmd.Flags().String(utils.NodeResourceGroupKey, "", "Node resource group name.")
+	vmmsCmd.Flags().String(utils.VMSSKey, "", "Virtual machine scale set name.")
+	vmmsCmd.Flags().String(utils.InstanceIDKey, "", "VM scale set instance ID.")
```

Comment:
I like the idea of dividing it in three sections: node, vmss and id. However, I was expecting to have only the three commands you wrote in the PR description: set/unset/show, and use a sort of key=value as argument. 

Considering the sections you are proposing, I suggest:

```bash
$ kubect-az config set vmss-instance.subscription "mySub"
$ kubect-az config set vmss-instance.node-resource-group "myNRG"
$ kubect-az config set vmss-instance.vmss "myVMSS"
$ kubect-az config set vmss-instance.id "myInstanceID"

$ kubect-az config set node.name "Node1"

$ kubect-az config set resource.id "12345..."
```

```bash
$ kubect-az config unset vmss.subscription

$ kubect-az config unset-all
```

```bash
$ kubect-az config show
vmss-instance:
	subscription: mySub
	node-resource-group: myNRG
	vmss: myVMSS
	id: myInstanceID
resource:
	id: /subscriptions/mySub/resourceGroups/myNRG/providers/Microsoft.Compute/virtualMachineScaleSets/myVMSS/virtualMachines/myInstanceID
node:
	name: Node1
```

Reply:
I was also thinking on the same terms and but then I saw how `kubectl config` has these `set-cluster/set-context/set-credentials` commands so I went to that route since it helps separating what is really needed for particular config to work. But I agree it might be simpler to just have one `set` command.

Reply:
Mmmm interesting. Let me check how `kubectl config` works. I was taking `git config` as example.

Reply:
I checked it and it saw it allows to do both: 

```bash
# set-cluster
$ k config set-cluster clustertest --server testserver

# set clusters.<name>.<property> <value>
$ k config set clusters.clustertest.server testserver2
```

You're right. I think we should follow that approach, it simplify the `vmss-instance` configuration. In the future maybe we can support multiple configurations like `kubectl config` does. 

So, I just suggest to change the naming to make it less redundant and don't have `set-node --node` but `set-node --name` and so on. I propose:

```bash
$ kubect-az config show
vmss-instance:
	subscription: mySub
	node-resource-group: myNRG
	vmss: myVMSS
	id: myInstanceID
resource:
	id: /subscriptions/mySub/resourceGroups/myNRG/providers/Microsoft.Compute/virtualMachineScaleSets/myVMSS/virtualMachines/myInstanceID
node:
	name: Node1
```

WDYT?

Reply:
> I think we should follow that approach, it simplify the vmss-instance configuration
:+1: 

> So, I just suggest to change the naming to make it less redundant and don't have set-node --node but set-node --name and so on

Agree. That makes it more clear. 

I like the idea about multiple named configs, it definitely helps easily switching context between nodes. We should add them in future or maybe in this PR as well :D